oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBConverter.java

```diff
@@ -0,0 +1,265 @@
+                    this.measureWrite.tag(fieldName, buildTag(fieldValue, columnSpec.getColumnClass()));
+                } else {
+                    this.measureWrite.field(fieldName, buildField(fieldValue, columnSpec.getColumnClass()));
+                }
+            } catch (BanyanDBException ex) {
+                log.error("fail to add tag", ex);
+            }
+        }
+
+        public void acceptID(String id) {
+            try {
+                this.measureWrite.setID(id);
+            } catch (BanyanDBException ex) {
+                log.error("fail to add ID tag", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, byte[] fieldValue) {
+            MetadataRegistry.ColumnSpec columnSpec = this.schema.getSpec(fieldName);
+            try {
+                if (columnSpec.getColumnType() == MetadataRegistry.ColumnType.TAG) {
+                    this.measureWrite.tag(fieldName, TagAndValue.binaryTagValue(fieldValue));
+                } else {
+                    this.measureWrite.field(fieldName, TagAndValue.binaryFieldValue(fieldValue));
+                }
+            } catch (BanyanDBException ex) {
+                log.error("fail to add binary tag/field", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, List<String> fieldValue) {
+            for (final String tagKeyAndValue : fieldValue) {
+                if (StringUtil.isEmpty(tagKeyAndValue)) {
+                    continue;
+                }
+                int pos = tagKeyAndValue.indexOf("=");
+                if (pos == -1) {
+                    continue;
+                }
+                String key = tagKeyAndValue.substring(0, pos);
+                String value = tagKeyAndValue.substring(pos + 1);
+                this.accept(key, value);
+            }
+        }
+
+        @Override
+        public Object get(String fieldName) {
+            return null;
+        }
+
+        @Override
+        public MeasureWrite obtain() {
+            return this.measureWrite;
+        }
+    }
+
+    private static Serializable<BanyandbModel.TagValue> buildTag(Object value, final Class<?> clazz) {
+        if (Integer.class.equals(clazz) || int.class.equals(clazz)) {
+            return TagAndValue.longTagValue(((Number) value).longValue());
+        } else if (Long.class.equals(clazz) || long.class.equals(clazz)) {
+            return TagAndValue.longTagValue((Long) value);
+        } else if (String.class.equals(clazz)) {
+            return TagAndValue.stringTagValue((String) value);
+        } else if (Double.class.equals(clazz) || double.class.equals(clazz)) {
+            return TagAndValue.binaryTagValue(ByteUtil.double2Bytes((double) value));
+        } else if (StorageDataComplexObject.class.isAssignableFrom(clazz)) {
+            return TagAndValue.stringTagValue(((StorageDataComplexObject<?>) value).toStorageData());
+        } else if (Layer.class.equals(clazz)) {
```

Comment:
We should not process `Layer`, you should check `enum`, if in that case, take this field as Int, and throw IllegalArgumentException when fail.

Reply:
Fixed. Use `Class.isEnum` to do the check.